aws/request.go

```diff
@@ -103,7 +103,7 @@ func New(cfg Config, metadata Metadata, handlers Handlers,
 	if err == nil {
 		// TODO so ugly
 		metadata.Endpoint = endpoint.URL
-		if len(endpoint.SigningName) > 0 {
+		if len(endpoint.SigningName) > 0 && !endpoint.SigningNameDerived {
```

Comment:
Can we just have the endpoint have a get signing name method that returns the correct signing name? I think that'll at least make it cleaner when dealing with cases where it isn't derived.

Reply:
because the signing name are modeled in both endpoint and service models, the SDK has to rely on both sources. In some cases the signing name is not included in the endpoint model,which is the default source.